components/log-ingestor/src/main.rs

```diff
@@ -1,3 +1,94 @@
-fn main() {
-    println!("Hello, world!");
+use anyhow::Context;
+use axum::{Router, routing::get};
+use clap::Parser;
+use clp_rust_utils::{clp_config::package, serde::yaml};
+use tracing_appender::rolling::{RollingFileAppender, Rotation};
+use tracing_subscriber::{self, fmt::writer::MakeWriterExt};
+
+fn read_config_and_credentials(
+    args: &Args,
+) -> anyhow::Result<(package::config::Config, package::credentials::Credentials)> {
+    let config_path = std::path::Path::new(args.config.as_str());
+    let config: package::config::Config = yaml::from_path(config_path).context(format!(
+        "Failed to load config file {}",
+        config_path.display()
+    ))?;
+
+    let credentials = package::credentials::Credentials {
+        database: package::credentials::Database {
+            password: secrecy::SecretString::new(
+                std::env::var("CLP_DB_PASS")
+                    .context("Expect `CLP_DB_PASS` env variable")?
+                    .into_boxed_str(),
+            ),
+            user: std::env::var("CLP_DB_USER").context("Expect `CLP_DB_USER` env variable")?,
+        },
+    };
+    Ok((config, credentials))
+}
+
+fn set_up_logging() -> anyhow::Result<tracing_appender::non_blocking::WorkerGuard> {
+    let logs_directory =
+        std::env::var("CLP_LOGS_DIR").context("Expect `CLP_LOGS_DIR` environment variable.")?;
+    let logs_directory = std::path::Path::new(logs_directory.as_str());
+    let file_appender =
+        RollingFileAppender::new(Rotation::HOURLY, logs_directory, "log_ingestor.log");
+    let (non_blocking_writer, guard) = tracing_appender::non_blocking(file_appender);
+    tracing_subscriber::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
```

Comment:
```suggestion
        .event_format(
            tracing_subscriber::fmt::format()
                .with_level(true)
                .with_target(false)
                .with_file(true)
                .with_line_number(true)
                .json(),
        )
        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
```
To be consistent with the latest API server setup.